src/components/fields/ArrayField.js

```diff
@@ -277,17 +277,16 @@ class ArrayField extends Component {
         }
       }
       onChange(
-        formData.map((item, i) => {
-          // i is string, index and newIndex are numbers,
-          // so using "==" to compare
-          if (i == newIndex) {
-            return formData[index];
-          } else if (i == index) {
-            return formData[newIndex];
-          } else {
-            return item;
-          }
-        }),
+        (function() {
+          // Copy item
+          let newFormData = formData.slice();
+
+          // Moves item from index to newIndex
+          newFormData.splice(index, 1);
+          newFormData.splice(newIndex, 0, formData[index]);
+
+          return newFormData;
+        })(),
```

Comment:
IIFE here is a bit ugly.

Reply:
Agree, was just trying to keep the same structure as before with an inline map-function. Moved out to its own function now